pylxd/container.py

```diff
@@ -30,6 +30,31 @@ class Container(mixin.Waitable, mixin.Marshallable):
     This class is not intended to be used directly, but rather to be used
     via `Client.containers.create`.
     """
+    class FilesManager(object):
+        """A pseudo-manager for namespacing file operations."""
+
+        def __init__(self, client, container):
+            self._client = client
+            self._container = container
+
+        def put(self, filepath, data):
+            response = self._client.api.containers[
+                self._container.name].files.post(
+                params={'path': filepath}, data=data)
+            return response.status_code == 200
+
+        def get(self, filepath):
+            response = self._client.api.containers[
+                self._container.name].files.get(
+                params={'path': filepath})
+            if response.status_code == 500:
+                # XXX: rockstar (15 Feb 2016) - This should really
+                # return a 404. I blame LXD.
```

Comment:
Can you file a bug against LXD when you see that kind of problem?

We indeed should be returning a 404 in such case, so it's probably best for your code to expect a 500 or 404 and have us fix the logic in LXD so it returns the right error code.